src/Corax/Querying/Matches/SortingMatches/SortingMultiMatch.Comparers.cs

```diff
@@ -402,14 +378,21 @@ public void Init(ref SortingMultiMatch<TInner> match, UnmanagedSpan<long> batchR
                 match._results.AddRange(batchResults);
                 return;
             }
+            // load terms for documents
+            _lookup.GetFor(batchResults, batchTermIds, BitConverter.DoubleToInt64Bits(double.MinValue));
 
-            _lookup.GetFor(batchResults, batchTermIds, long.MinValue);
-            match._token.ThrowIfCancellationRequested();
-            var indexes = EntryComparerHelper.NumericSortBatch(ref match, batchTermIds, batchTerms, new EntryComparerByLong(), comparer2, comparer3);
+            var heapSize = Math.Min(match._take, batchResults.Length);
+            heapSize = heapSize < 0 ? batchResults.Length : heapSize;
```

Comment:
How / when `heapSize` can be < 0?

Reply:
I see below that `match._take` can be `-1`. A bit tricky but I believe it's okay here